src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -575,6 +594,10 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
                         int reasonPhraseEnd = newLineCharIndex >= 0 ? newLineCharIndex : responseHeaderLength;
                         response.ReasonPhrase = responseHeader.Substring(reasonPhraseIndex, reasonPhraseEnd - reasonPhraseIndex);
                     }
+                    state._isRedirect = state._handler.AutomaticRedirection &&
+                         (response.StatusCode == HttpStatusCode.Redirect ||
+                         response.StatusCode == HttpStatusCode.RedirectKeepVerb ||
+                         response.StatusCode == HttpStatusCode.RedirectMethod) ;
```

Comment:
It seems like when the second request comes back, we'll end up setting _isRedirect back to false.  Is that correct?